Tests/SwiftSyntaxTest/SymbolNameTests.swift

```diff
@@ -0,0 +1,358 @@
+  }
+
+  func testSubscriptNameWithMultipleNamedParameters() {
+    assertSymbolName(ofSubscript: "subscript(x: Int, y: Int) -> Int { 0 }", expecting: "subscript(_:_:)")
+  }
+
+  func testSubscriptNameWithMultipleParametersAndExternalNames() {
+    assertSymbolName(
+      ofSubscript: "subscript(indexX x: Int, indexY y: Int) -> Int { 0 }",
+      expecting: "subscript(indexX:indexY:)"
+    )
+  }
+
+  func testSubscriptNameWithParameterNameInBackticks() {
+    assertSymbolName(ofSubscript: "subscript(`index` i: Int) -> Int { 0 }", expecting: "subscript(`index`:)")
+  }
+
+  private func assertSymbolName(
+    ofSubscript subscriptDeclaration: DeclSyntax,
+    expecting expectedSymbolName: String,
+    file: StaticString = #filePath,
+    line: UInt = #line
+  ) {
+    guard let subscriptDecl = subscriptDeclaration.as(SubscriptDeclSyntax.self) else {
+      XCTFail("Expected subscript declaration not found.", file: file, line: line)
+      return
+    }
+
+    XCTAssertEqual(subscriptDecl.symbolName, expectedSymbolName, file: file, line: line)
+  }
+
+  // MARK: - Enum Case Element
+
+  func testEnumCaseElementWithoutAssociatedValues() {
+    assertSymbolName(ofEnumCaseElementFromEnumCase: "case a", expecting: "a")
+  }
+
+  func testEnumCaseElementWithSingleNamedAssociatedValue() {
+    assertSymbolName(ofEnumCaseElementFromEnumCase: "case b(c: Int)", expecting: "b(c:)")
+  }
+
+  func testEnumCaseElementWithSingleUnnamedAssociatedValue() {
+    assertSymbolName(ofEnumCaseElementFromEnumCase: "case d(Int)", expecting: "d(_:)")
+  }
+
+  func testEnumCaseElementWithSingleUnnamedAssociatedValueWithUnderscore() {
+    assertSymbolName(ofEnumCaseElementFromEnumCase: "case e(_: Int)", expecting: "e(_:)")
+  }
+
+  func testEnumCaseElementWithMultipleNamedAssociatedValues() {
+    assertSymbolName(ofEnumCaseElementFromEnumCase: "case f(g: Int, h: Int)", expecting: "f(g:h:)")
+  }
+
+  func testEnumCaseElementNameWithBackticks() {
+    assertSymbolName(ofEnumCaseElementFromEnumCase: "case `i`", expecting: "`i`")
+  }
+
+  func testEnumCaseElementWithAssociatedValueNameInBackticks() {
+    assertSymbolName(ofEnumCaseElementFromEnumCase: "case j(`k`: Int)", expecting: "j(`k`:)")
+  }
+
+  private func assertSymbolName(
+    ofEnumCaseElementFromEnumCase enumCaseDeclaration: DeclSyntax,
+    expecting expectedSymbolName: String,
+    file: StaticString = #filePath,
+    line: UInt = #line
+  ) {
+    guard let enumCaseDecl = enumCaseDeclaration.as(EnumCaseDeclSyntax.self),
+      let enumCaseElement = enumCaseDecl.elements.first
+    else {
+      XCTFail("Expected enum case element declaration not found.", file: file, line: line)
+      return
+    }
+    XCTAssertEqual(enumCaseElement.symbolName, expectedSymbolName, file: file, line: line)
+  }
+
+  // MARK: - Macro
+
+  func testMacroDeclWithoutParameters() {
+    assertSymbolName(
+      ofMacroDecl: ##"macro myMacro() = #externalMacro"##,
```

Comment:
I don’t think we need the `= #externalMacro` part for the test. 

And I think normal string literals without `##` should be sufficient.